scripts/functions.sh

```diff
@@ -0,0 +1,99 @@
+# ------------------------------------------------------------------------------
+# This file is to be included by other scripts that want to include functionality exported below.
+#
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2019 solidity contributors.
+#------------------------------------------------------------------------------
+
+if [[ "$CIRCLECI" ]]; then
+    function printTask() { echo "$(tput bold)$(tput setaf 2)$1$(tput setaf 7)"; }
+    function printError() { echo "$(tput setaf 1)$1$(tput setaf 7)"; }
+else
+    function printTask() { echo "$(tput bold)$(tput setaf 2)$1$(tput sgr0)"; }
+    function printError() { echo "$(tput setaf 1)$1$(tput sgr0)"; }
+fi
+
+function find_valgrind()
+{
+    local path=`which valgrind 2>/dev/null`
+    echo -n "${path}"
+}
+
+function get_valgrind_prefix()
+{
+    if [[ "${ENABLE_VALGRIND}" != "yes" ]]; then
+        return
+    fi
+    if [[ "$CIRCLECI" ]]; then
+        apt-get install -qqy valgrind
+    fi
+    local exe=`find_valgrind`
```

Comment:
Why not just keep the shell doing its job in discovering the location of executables?